tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/SelectedFunctionTest.java

```diff
@@ -0,0 +1,75 @@
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.testerina.test;
+
+import org.ballerinalang.test.context.BMainInstance;
+import org.ballerinalang.test.context.BallerinaTestException;
+import org.ballerinalang.test.context.LogLeecher;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/**
+ * Test class containing tests related to selective function tests.
+ */
+public class SelectedFunctionTest extends BaseTestCase {
+
+    private BMainInstance balClient;
+    private String projectPath;
+
+    @BeforeClass
+    public void setup() throws BallerinaTestException {
+        balClient = new BMainInstance(balServer);
+        projectPath = basicTestsProjectPath.toString();
+    }
+
+    @Test
+    public void testSingleFunctionExecution() throws BallerinaTestException {
+        String msg = "1 passing";
+        LogLeecher clientLeecher = new LogLeecher(msg);
+        balClient.runMain("test", new String[]{"--functions", "testFunc", "beforeEachAfterEach"},
+                null, new String[]{}, new LogLeecher[]{clientLeecher}, projectPath);
+        clientLeecher.waitForText(20000);
+    }
+
+    @Test
+    public void testDependentFunctionExecution() throws BallerinaTestException {
+        String msg = "2 passing";
+        LogLeecher clientLeecher = new LogLeecher(msg);
+        balClient.runMain("test", new String[]{"--functions", "testFunc2", "beforeEachAfterEach"},
+                null, new String[]{}, new LogLeecher[]{clientLeecher}, projectPath);
+        clientLeecher.waitForText(20000);
+    }
+
+    @Test
+    public void testMultipleFunctionExecution() throws BallerinaTestException {
+        String msg = "2 passing";
+        LogLeecher clientLeecher = new LogLeecher(msg);
+        balClient.runMain("test", new String[]{"--functions", "testFunc,testFunc2",
+                "beforeEachAfterEach"}, null, new String[]{}, new LogLeecher[]{clientLeecher}, projectPath);
+        clientLeecher.waitForText(20000);
+    }
+
+    @Test
+    public void testNonExistingFunctionExecution() throws BallerinaTestException {
+        String msg = "No tests found";
+        LogLeecher clientLeecher = new LogLeecher(msg);
+        balClient.runMain("test", new String[]{"--functions", "nonExistingFunc", "--all"},
+                null, new String[]{}, new LogLeecher[]{clientLeecher}, projectPath);
+        clientLeecher.waitForText(20000);
```

Comment:
What is the error code of this.

Reply:
If the provided text is not found within the timeout period, an error message as "Timeout expired waiting for matching log" will be given. Does that answer your question?

Reply:
AFAIU, @hevayo is asking about the exit code of the command. According to the current implementation, the exit code is `0`, when no tests found to execute. Ideally this should be `1`.

Reply:
Yeh sorry I was referring to exit code